File: hackerrank/day8/day8.js
'use strict';

const fs = require('fs');

const input = fs.readFileSync(0, 'utf8').trim().split('\n');

const n = parseInt(input[0]);

const phoneBook = {};

for (let i = 1; i <= n; i++) {
    const [name, number] = input[i].split(' ');
    phoneBook[name] = number;
}

for (let i = n + 1; i < input.length; i++) {
    const query = input[i];

    if (phoneBook.hasOwnProperty(query)) {
        console.log(`${query}=${phoneBook[query]}`);
    } else {
        console.log("Not found");
    }
}

File: hackerrank/day8/day8.cs
using System;
using System.Collections.Generic;

class Solution
{
    static void Main(string[] args)
    {
        int n = int.Parse(Console.ReadLine());

        Dictionary<string, string> phoneBook = new Dictionary<string, string>();

        for (int i = 0; i < n; i++)
        {
            string[] data = Console.ReadLine().Split();
            phoneBook[data[0]] = data[1];
        }

        string query;

        while ((query = Console.ReadLine()) != null)
        {
            if (phoneBook.ContainsKey(query))
            {
                Console.WriteLine($"{query}={phoneBook[query]}");
            }
            else
            {
                Console.WriteLine("Not found");
            }
        }
    }
}
